cores/arduino/HardwareTimer.cpp

```diff
@@ -913,6 +910,10 @@ void HardwareTimer::setPWM(uint32_t channel, PinName pin, uint32_t frequency, ui
   */
 void HardwareTimer::setInterruptPriority(uint32_t preemptPriority, uint32_t subPriority)
 {
+  // Set priority for immediate use
+  NVIC_SetPriority(getTimerUpIrq(_timerObj.handle.Instance), NVIC_EncodePriority(NVIC_GetPriorityGrouping(), preemptPriority, subPriority));
+
```

Comment:
IRQ number may be different for Update interrupt and Capture/Compare interrupt (depending on timer instance). Thus it may be necessary to update also Caputure/Compare priority. Add something like this:
```suggestion
if (getTimerCCIrq(_timerObj.handle.Instance) != getTimerUpIrq(_timerObj.handle.Instance)) {
  NVIC_SetPriority(getTimerCCIrq(_timerObj.handle.Instance), NVIC_EncodePriority(NVIC_GetPriorityGrouping(), preemptPriority, subPriority));
}
```

Reply:
I made this update, and also replace NVIC_SetPriority with HAL_NVIC_SetPriority. It's not entirely clear to me which entrypoint I should be using from here, but the HAL_ version took care of encoding the priority for me.